### PR TITLE
[ticket/14748] Make sure config values are casted to int

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [ ] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-14748

PHPBB3-14748

### CODE REVIEW

**Comment by marc1706** [comment]
Remaining failures seem to be caused by a bug in PHP 7.1:
https://3v4l.org/nZEP9

**Comment by marc1706** [comment]
This might actually finally be correct behavior as it does no longer ignore the backslash or no longer allows it inside the userinfo part.

File: phpBB/phpbb/db/migration/data/v310/timezone.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ public function update_timezones($start)
 	*/
 	public function convert_phpbb30_timezone($timezone, $dst)
 	{
-		$offset = $timezone + $dst;
+		$offset = (int) $timezone + (int) $dst;
```
Comment (marc1706) [inline phpBB/phpbb/db/migration/data/v310/timezone.php]: `(float)`

**Comment by marc1706** [comment]
Closing in favor of #4457

**Comment by rxu** [comment]
Regarding remaining failures, in PHP 7.1+ ```parse_url()``` was changed to be more restrictive and supports RFC3986 now, so it just returns false on urls like ```http://localhost.foobar.com\\@localhost/troll/http://localhost/``` (from our tests). That's why tests fail on expecting exceptions from ```redirect()``` because ```redirect()``` returns url to index page.
We might want to throw exception in that case as well for consistency instead of redirecting to the same page.

```parse_url()``` output in PHP 7.1+ and older versions:
parse_url

**Pull request closed without merging**